# **Aula 09**- Exercícios

07-10-2026

## **Atividade 01:**

Uma empresa de comércio eletrônico deseja analisar o desempenho de suas vendas durante uma semana.
Foram registrados, para cada dia, os seguintes indicadores:

* quantidade de pedidos;
* faturamento (R$);
* número de clientes;
* valor médio dos pedidos;
* quantidade de reclamações.


Utilize para organizar, analisar e interpretar os dados.

A estrutura das colunas de dados é: **[Pedidos, Faturamento, Clientes, Ticket_Medio, Reclamações]**

### Parte 0 — Declarando os dados

In [28]:
# Importanto a biblioteca
import numpy as np

# evita notação científica
np.set_printoptions(suppress=True)

# Colunas: [Pedidos, Faturamento, Clientes, Ticket_Medio, Reclamações]
dados = np.array([
  [120, 15000, 100, 125.00, 4],
  [135, 16200, 108, 120.00, 5],
  [98, 11700, 82, 119.39, 3],
  [150, 21000, 125, 140.00, 7],
  [175, 24500, 140, 140.00, 9],
  [160, 22400, 132, 140.00, 6],
  [110, 13200, 95, 120.00, 2]
])

dias = np.array([
  "Segunda",
  "Terça",
  "Quarta",
  "Quinta",
  "Sexta",
  "Sábado",
  "Domingo"
])

# Atalhos: colunas de pedidos, faturamento e reclamações (um valor por dia)
ped, fat, recl = dados[:, 0], dados[:, 1], dados[:, 4]

### Parte 1 — Conhecendo os dados

In [29]:
print("Linhas:", dados.shape[0])
print("Colunas:", dados.shape[1])
print("Total de valores:", dados.size)
print("Dimensões:", dados.ndim)
print("Tipo:", dados.dtype)

Linhas: 7
Colunas: 5
Total de valores: 35
Dimensões: 2
Tipo: float64


### Parte 2 — Acessando os dados

In [30]:
print("2.1 Sexta:", dados[4])
print("2.2 Faturamento:", dados[:, 1])
print("2.3 Reclamações:", dados[:, 4]) # todas as linhas, coluna 4
print("2.4 Quinta a Domingo:\n", dados[3:]) # slicing: da linha 3 até o fim
print("2.5 Pedidos, faturamento, clientes:\n", dados[:, :3]) # colunas 0 a 2

2.1 Sexta: [  175. 24500.   140.   140.     9.]
2.2 Faturamento: [15000. 16200. 11700. 21000. 24500. 22400. 13200.]
2.3 Reclamações: [4. 5. 3. 7. 9. 6. 2.]
2.4 Quinta a Domingo:
 [[  150. 21000.   125.   140.     7.]
 [  175. 24500.   140.   140.     9.]
 [  160. 22400.   132.   140.     6.]
 [  110. 13200.    95.   120.     2.]]
2.5 Pedidos, faturamento, clientes:
 [[  120. 15000.   100.]
 [  135. 16200.   108.]
 [   98. 11700.    82.]
 [  150. 21000.   125.]
 [  175. 24500.   140.]
 [  160. 22400.   132.]
 [  110. 13200.    95.]]


### Parte 3 — Operações numéricas

In [31]:
# np.sum e np.mean aplicados sobre uma coluna
print("3.1 Total de pedidos:", int(np.sum(dados[:, 0])))
print("3.2 Faturamento total: R$", np.sum(dados[:, 1]))
print("3.3 Total de clientes:", int(np.sum(dados[:, 2])))
print("3.4 Média diária de pedidos:", round(np.mean(dados[:, 0]), 2))
print("3.5 Média diária de faturamento: R$", round(np.mean(dados[:, 1]), 2))
print("3.6 Média de reclamações/dia:", round(np.mean(dados[:, 4]), 2))

3.1 Total de pedidos: 948
3.2 Faturamento total: R$ 124000.0
3.3 Total de clientes: 782
3.4 Média diária de pedidos: 135.43
3.5 Média diária de faturamento: R$ 17714.29
3.6 Média de reclamações/dia: 5.14


### Parte 4 — Análise por dia

In [32]:
# argmax/argmin devolvem a POSIÇÃO do maior/menor valor; usamos essa posição para obter o nome do dia
print("4.1 Mais pedidos:", dias[np.argmax(ped)], "-", int(np.max(ped)))
print("4.1 Menos pedidos:", dias[np.argmin(ped)], "-", int(np.min(ped)))
print("4.2 Maior faturamento:", dias[np.argmax(fat)], "- R$", np.max(fat))
print("4.2 Menor faturamento:", dias[np.argmin(fat)], "- R$", np.min(fat))

# amplitude = maior valor - menor valor
print("4.3 Maior: R$", np.max(fat), "| Menor: R$", np.min(fat), "| Amplitude: R$", np.max(fat) - np.min(fat))

4.1 Mais pedidos: Sexta - 175
4.1 Menos pedidos: Quarta - 98
4.2 Maior faturamento: Sexta - R$ 24500.0
4.2 Menor faturamento: Quarta - R$ 11700.0
4.3 Maior: R$ 24500.0 | Menor: R$ 11700.0 | Amplitude: R$ 12800.0


### Parte 5 — Máscaras e filtros

In [39]:
# máscara booleana (ex.: fat > 20000) usada como índice para filtrar os dias
print("5.1 Dias faturamento > 20.000:", dias[fat > 20000])
print("5.2 Dias com mais de 150 pedidos:", dias[ped > 150])
print("5.3 Dias com mais de 5 reclamações:", dias[recl > 5])
print("5.4 Faturamento dos dias com pedidos > 150:", fat[ped > 150])

# soma de valores True = contagem de dias que atendem à condição
print("5.5 Dias faturamento abaixo da média semanal:", np.sum(fat < np.mean(fat)))

5.1 Dias faturamento > 20.000: ['Quinta' 'Sexta' 'Sábado']
5.2 Dias com mais de 150 pedidos: ['Sexta' 'Sábado']
5.3 Dias com mais de 5 reclamações: ['Quinta' 'Sexta' 'Sábado']
5.4 Faturamento dos dias com pedidos > 150: [24500. 22400.]
5.5 Dias faturamento abaixo da média semanal: 4


### Parte 6 — Classificação dos dias

In [34]:
# np.where(condição, valor_se_True, valor_se_False) classifica todos os dias de uma vez
classificacao = np.where(fat >= 20000, "Alto Desempenho", "Desempenho Normal")

print("6.1", np.column_stack((dias, classificacao)))
print("6.2 Dias de classificação Alto Desempenho:", np.sum(classificacao == "Alto Desempenho"))

6.1 [['Segunda' 'Desempenho Normal']
 ['Terça' 'Desempenho Normal']
 ['Quarta' 'Desempenho Normal']
 ['Quinta' 'Alto Desempenho']
 ['Sexta' 'Alto Desempenho']
 ['Sábado' 'Alto Desempenho']
 ['Domingo' 'Desempenho Normal']]
6.2 Dias de classificação Alto Desempenho: 3


### Parte 7 — Estatística

In [35]:
# medidas estatísticas do faturamento
media, mediana, desvio = np.mean(fat), np.median(fat), np.std(fat)

print(f"Média: R$ {media:.2f}")
print(f"Mediana: R$ {mediana:.2f}")
print(f"Desvio padrão: R$ {desvio:.2f}")
print(f"Mínimo: R$ {np.min(fat):.2f}")
print(f"Máximo: R$ {np.max(fat):.2f}")

Média: R$ 17714.29
Mediana: R$ 16200.00
Desvio padrão: R$ 4551.43
Mínimo: R$ 11700.00
Máximo: R$ 24500.00


***O faturamento apresenta grande variação ao longo da semana? Justifique utilizando os valores calculados.***

**Resposta:** Sim, a variação é considerável. O desvio padrão de 4.551 equivale a cerca de 26% da média de 17.700 e a amplitude, 12.800, é superior ao próprio faturamento mínimo. Os dias de maior movimento, quinta a sábado, puxam a média para cima.

### Parte 8 — Trabalhando com axis

In [36]:
# axis=0 -> calcula ao longo das linhas, gerando um resultado por coluna (indicador)
nomes = ["Pedidos", "Faturamento", "Clientes", "Ticket", "Reclamações"]
medias = np.mean(dados, axis=0)
maximos = np.max(dados, axis=0)
minimos = np.min(dados, axis=0)

print("8.1 Médias :", np.round(medias, 2))
print("8.2 Máximos:", maximos)
print("8.3 Mínimos:", minimos)

# variação relativa = (máx - mín) / média; permite comparar indicadores de escalas diferentes
var_rel = (maximos - minimos) / medias
print("8.4 Variação relativa:", np.round(var_rel, 2))
print("Indicador com maior variação relativa:", nomes[np.argmax(var_rel)])

8.1 Médias : [  135.43 17714.29   111.71   129.2      5.14]
8.2 Máximos: [  175. 24500.   140.   140.     9.]
8.3 Mínimos: [   98.   11700.      82.     119.39     2.  ]
8.4 Variação relativa: [0.57 0.72 0.52 0.16 1.36]
Indicador com maior variação relativa: Reclamações


### Parte 9 — Análise integrada

In [37]:
# i = posição do dia com maior faturamento; comparamos com a posição do maior nº de pedidos/reclamações
i = np.argmax(fat)
print("9.1 Melhor dia considerando o faturamento:", dias[i])
print("9.2 Também teve mais pedidos?", i == np.argmax(ped))
print("9.3 Também teve mais reclamações?", i == np.argmax(recl))

# "&" combina duas máscaras: os dois critérios precisam ser verdadeiros no mesmo dia
mascara = (ped > 150) & (fat > 20000)
print("9.4 Dias com >150 pedidos e >R$20.000:", np.sum(mascara), "=", dias[mascara])

9.1 Melhor dia considerando o faturamento: Sexta
9.2 Também teve mais pedidos? True
9.3 Também teve mais reclamações? True
9.4 Dias com >150 pedidos e >R$20.000: 2 = ['Sexta' 'Sábado']


### Parte 10 — Desafio de análise

In [38]:
print("----- RESUMO DA SEMANA -----")
print(f"Total de pedidos: {int(np.sum(ped))}")
print(f"Faturamento total: R$ {np.sum(fat):,.2f}")
print(f"Média diária de pedidos: {np.mean(ped):.2f}")
print(f"Média diária de faturamento: R$ {np.mean(fat):,.2f}")
print(f"Melhor dia: {dias[np.argmax(fat)]}")
print(f"Pior dia: {dias[np.argmin(fat)]}")
print(f"Maior número de reclamações: {int(np.max(recl))} ({dias[np.argmax(recl)]})")
print(f"Dias de alto desempenho: {np.sum(fat >= 20000)} = {dias[fat >= 20000]}")

----- RESUMO DA SEMANA -----
Total de pedidos: 948
Faturamento total: R$ 124,000.00
Média diária de pedidos: 135.43
Média diária de faturamento: R$ 17,714.29
Melhor dia: Sexta
Pior dia: Quarta
Maior número de reclamações: 9 (Sexta)
Dias de alto desempenho: 3 = ['Quinta' 'Sexta' 'Sábado']


### Parte 11 — Conclusão

**Desempenho geral:** a semana somou 948 pedidos e 124.000 de faturamento, com média de ~135 pedidos e ~17.700 de faturamento por dia. O desempenho foi concentrado: quinta, sexta e sábado superaram 20.000 (alto desempenho), com **sexta-feira** como melhor dia com 175 pedidos = 24.500.

**Dias que merecem atenção:**
- **Quarta-feira:** pior dia, somente 98 pedidos = 11.700.
- **Sexta e quinta:** maiores volumes de reclamações: 9 e 7 respectivamente.
- **Domingo e segunda:** faturamento baixo, mas com poucas reclamações.

**Relação entre os indicadores:** mais pedidos geram mais faturamento e também mais reclamações. Isso sugere que as reclamações crescem junto com o volume de pedidos.